raiden-ts/src/transport/epics.ts

```diff
@@ -863,14 +898,21 @@ export const matrixMessageReceivedEpic = (
             (event, room) => ({ matrix, event, room }),
           ),
         ),
-        // filter for text messages from other users
+        withLatestFrom(config$),
+        // filter for text messages not from us and not from global rooms
         filter(
-          ({ event, matrix }) =>
+          ([{ matrix, event, room }, config]) =>
             event.getType() === 'm.room.message' &&
+            event.event.content.msgtype === 'm.text' &&
             event.getSender() !== matrix.getUserId() &&
-            event.event.content.msgtype === 'm.text',
+            ![config.discoveryRoom, config.pfsRoom]
```

Comment:
We do this quite often in this PR with the same params. Might make sense to have it in its own function somewhere